parquet/src/main/java/org/apache/iceberg/parquet/ParquetDictionaryRowGroupFilter.java

```diff
@@ -148,6 +148,16 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       return ROWS_MIGHT_MATCH;
     }
 
+    @Override
+    public <T> Boolean isNaN(BoundReference<T> ref) {
+      return ROWS_MIGHT_MATCH;
```

Comment:
If there is a NaN value, then it would be in the Parquet dictionary. I think that we can implement this for Parquet.

Reply:
Thanks! I'll see how it works